profiles/lib/stringlist.lua

```diff
@@ -0,0 +1,28 @@
+local Stringlist = {}
+
+-- Takes a string representing a list separated by delim and canonicalizes containing spaces.
+--
+-- Example:
+-- In: aaa;bbb; ccc;  d;dd"
+-- Out: aaa; bbb; ccc; d; dd"
+function Stringlist.canonicalize_spaces(str, delim)
```

Comment:
Devils advocate here: Why not do this in C++-land and expose a helper function to lua?